stroom-search/stroom-search-elastic/src/main/java/stroom/search/elastic/suggest/ElasticSuggestionsQueryHandlerImpl.java

```diff
@@ -0,0 +1,118 @@
+package stroom.search.elastic.suggest;
+
+import stroom.datasource.api.v2.AbstractField;
+import stroom.datasource.api.v2.FieldTypes;
+import stroom.query.shared.FetchSuggestionsRequest;
+import stroom.query.util.LambdaLogger;
+import stroom.query.util.LambdaLoggerFactory;
+import stroom.search.elastic.ElasticClientCache;
+import stroom.search.elastic.ElasticClusterStore;
+import stroom.search.elastic.ElasticIndexStore;
+import stroom.search.elastic.shared.ElasticClusterDoc;
+import stroom.search.elastic.shared.ElasticIndexDoc;
+import stroom.task.api.TaskContextFactory;
+
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.elasticsearch.search.suggest.Suggest;
+import org.elasticsearch.search.suggest.SuggestBuilder;
+import org.elasticsearch.search.suggest.SuggestBuilders;
+import org.elasticsearch.search.suggest.SuggestionBuilder;
+import org.elasticsearch.search.suggest.term.TermSuggestion;
+import org.elasticsearch.search.suggest.term.TermSuggestionBuilder;
+import org.elasticsearch.search.suggest.term.TermSuggestionBuilder.SuggestMode;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@Singleton
+public class ElasticSuggestionsQueryHandlerImpl implements ElasticSuggestionsQueryHandler {
+
+    private static final LambdaLogger LOGGER = LambdaLoggerFactory.getLogger(ElasticSuggestionsQueryHandlerImpl.class);
+
+    private final ElasticClientCache elasticClientCache;
+    private final ElasticClusterStore elasticClusterStore;
+    private final ElasticIndexStore elasticIndexStore;
+    private final ElasticSuggestConfig elasticSuggestConfig;
+    private final TaskContextFactory taskContextFactory;
+
+    @Inject
+    public ElasticSuggestionsQueryHandlerImpl(final ElasticClientCache elasticClientCache,
+                                              final ElasticClusterStore elasticClusterStore,
+                                              final ElasticIndexStore elasticIndexStore,
+                                              final ElasticSuggestConfig elasticSuggestConfig,
```

Comment:
Needs to be injected as a provider then call .get() on each use.  This ensures you have the latest instance if the config has changed.